packages/botfuel-dialog/src/dialogs/prompt-dialog.js

```diff
@@ -216,13 +247,37 @@ class PromptDialog extends Dialog {
           missingEntities: newMissingEntities,
         };
       },
+      // initial state
       {
         matchedEntities: previouslyMatchedEntities,
         remainingCandidates: candidates,
-        missingEntities: parameters,
+        missingEntities: entities,
       },
     );
-    return { matchedEntities: result.matchedEntities, missingEntities: result.missingEntities };
+    return {
+      matchedEntities: result.matchedEntities,
+      missingEntities: result.missingEntities,
+    };
+  }
+
+  /**
+   * Compute sorted question entities as a Map
+   * For PromptDialog this is the same as the missingEntities ordered by priority
+   * Override this function if you want to modify the entities to be asked next
+   * like in a Faceted Search for example
+   * @param {Object} matchedEntities - matched entities
```

Comment:
why do we need this?

Reply:
faceted search need to query from matched entities to see if some missing entities can be deduced. See search-dialog here https://github.com/Botfuel/botfuel-facetedsearch